tests/multigrid/test_generalised_hierarchy.py

```diff
@@ -0,0 +1,175 @@
+from firedrake import *
+from firedrake.mg import *
+from firedrake.mg.utils import *
+
+import numpy as np
+
+import pytest
+
+
+def test_generalised_mesh_hierarchy():
+
+    mesh = UnitSquareMesh(5, 5)
+    L = 4
+    GMH = GeneralisedMeshHierarchy(mesh, L)
+
+    assert len(GMH) == L + 1
+
+    assert GMH.M == 2
+
+    a = 0
+    try:
+        GMH = GeneralisedMeshHierarchy(mesh, L, 3)
+
+    except ValueError:
+        a = 1
+
+    assert a == 1
+
+    L = 3
+    M = 4
+    GMH = GeneralisedMeshHierarchy(mesh, L, M)
+
+    assert len(GMH) == L + 1
+
+
+def test_attribute():
+
+    mesh = UnitSquareMesh(5, 5)
+    L = 3
+    M = 4
+    GMH = GeneralisedMeshHierarchy(mesh, L, M)
+
+    assert hasattr(GMH, '_full_hierarchy') == 1
+
+    GFSH = GeneralisedFunctionSpaceHierarchy(GMH, 'DG', 0)
+
+    assert hasattr(GFSH, '_full_hierarchy') == 1
+
+    F = Function(GFSH[0])
+
+    assert isinstance(get_level(F.function_space())[0], FunctionSpaceHierarchy) == 1
```

Comment:
Wat?  `assert isinstance(..., FunctionSpaceHierarchy)`.  What is the `== 1` checking?